upstream-sha256: b5cfae42725ed2ebbc5cf2e7c4700fa90fdbad851c01c31a600dfe4917244845
actual-sha256: 77de528c42c005cfc758b2e4bcc1eb375b4838300d4842d5d38969132b0f4d2f
--- upstream.expected
+++ tolk.actual
@@ -20,7 +20,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3));
+  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
 }
 ---
 typedef long unsigned int size_t;
@@ -45,7 +45,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = (2.0f*((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
+  *(data0_16+(lidx1+alu0)) = (2.0f*((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3))));
 }
 ---
 typedef long unsigned int size_t;
@@ -62,7 +62,7 @@
   float val0 = (*(data1_16+alu1));
   float4 val1 = (*((float4*)((data2_16+alu0))));
   float4 val2 = (*((float4*)((data3_16+(lidx1<<2)))));
-  *(data0_16+alu1) = ((2.0f*val0)+(val1.x*val2.x)+(val1.y*val2.y)+(val1.z*val2.z)+(val1.w*val2.w));
+  *(data0_16+alu1) = ((2.0f*val0)+(0.0f+(val1.x*val2.x)+(val1.y*val2.y)+(val1.z*val2.z)+(val1.w*val2.w)));
 }
 ---
 typedef long unsigned int size_t;
@@ -88,7 +88,7 @@
   float val7 = (*(data4_16+(lidx1+12)));
   float val8 = (*(data4_16+(lidx1+4)));
   float val9 = (*(data4_16+(lidx1+8)));
-  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7));
+  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(0.0f+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7)));
 }
 ---
 typedef long unsigned int size_t;
@@ -114,7 +114,7 @@
   float val7 = (*(data4_16+(lidx1+12)));
   float val8 = (*(data4_16+(lidx1+4)));
   float val9 = (*(data4_16+(lidx1+8)));
-  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7));
+  *(data0_16+(lidx1+(lidx0<<2))) = ((*(buf0+0))+(0.0f+(val2*val6)+(val4*val8)+(val5*val9)+(val3*val7)));
 }
 ---
 typedef long unsigned int size_t;
@@ -129,5 +129,5 @@
   float4 val0 = (*((float4*)((data1_16+alu0))));
   int lidx1 = __ockl_get_local_id(1); /* 4 */
   float4 val1 = (*((float4*)((data2_16+(lidx1<<2)))));
-  *(data0_16+(lidx1+alu0)) = ((val0.x*val1.x)+(val0.y*val1.y)+(val0.z*val1.z)+(val0.w*val1.w));
+  *(data0_16+(lidx1+alu0)) = (0.0f+(val0.x*val1.x)+(val0.y*val1.y)+(val0.z*val1.z)+(val0.w*val1.w));
 }
